src/app/guards/user.guard.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+
+@Injectable()
+export class UserGuard implements CanActivate {
+  skip: boolean;
+  constructor(private router: Router) { }
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    const isAdmin = JSON.parse(sessionStorage.getItem('isAdmin'));
+    if (isAdmin) {
+      // logged in so return true
+      return true;
+    }
+    // not logged in so redirect to login page with the return url
```

Comment:
what if optional?

Reply:
This is to check if user is admin. In all other case it will return false.